### PR TITLE
ERMS-2360, ERMS-2515,ERMS-2357

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: app/grails-app/services/de/laser/SurveyUpdateService.groovy [hunk 1]
```diff
@@ -104,65 +104,85 @@ class SurveyUpdateService extends AbstractLockableService {
...
-                        mailService.sendMail {
-                            to      emailReceiver
-                            from    from
-                            cc      ccAddress
-                            replyTo replyTo
-                            subject mailSubject
-                            body    (view: "/mailTemplates/text/notificationSurvey", model: [user: user, org: org, survey: survey])
+                        List generalContactsEMails = []
+
+                        survey.owner.getGeneralContactPersons(false)?.each { person ->
```
Comment (croqueGrec09) [inline app/grails-app/services/de/laser/SurveyUpdateService.groovy:146]: If block would be more readable

File: app/grails-app/services/de/laser/SurveyUpdateService.groovy [hunk 2]
```diff
@@ -104,65 +104,85 @@ class SurveyUpdateService extends AbstractLockableService {
...
+                            person?.contacts?.each { contact ->
```
Comment (croqueGrec09) [inline app/grails-app/services/de/laser/SurveyUpdateService.groovy:147]: person should not be null here, make contacts in if-block as well

File: app/grails-app/services/de/laser/SurveyUpdateService.groovy [hunk 3]
```diff
@@ -104,65 +104,85 @@ class SurveyUpdateService extends AbstractLockableService {
 
     }
 
+    def emailsToSurveyUsersOfOrg(SurveyInfo surveyInfo, Org org){
+
+            //Only User that approved
+            List<UserOrg> userOrgs = UserOrg.findAllByOrgAndStatus(org, 1)
```
Comment (klodav) [inline app/grails-app/services/de/laser/SurveyUpdateService.groovy:110]: use constants: UserOrg.STATUS_APPROVED

File: app/grails-app/services/de/laser/SurveyUpdateService.groovy [hunk 4]
```diff
@@ -104,65 +104,85 @@ class SurveyUpdateService extends AbstractLockableService {
...
+
+            //Only User with Notification by Email and for Surveys Start
+            userOrgs.each { userOrg ->
+                if(userOrg.user?.getSettingsValue(UserSettings.KEYS.IS_NOTIFICATION_FOR_SURVEYS_START) == RDStore.YN_YES &&
```
Comment (klodav) [inline app/grails-app/services/de/laser/SurveyUpdateService.groovy:114]: userOrg.user is never null

**Pull request merged**